File: ya/service.py
"""Shared task execution used by the command line and native GUI clients."""

from __future__ import annotations

from typing import Callable

from .config import ModelConfig
from .deepseek import DeepSeekClient
from .orchestrator import RunResult, single_agent, toa_agent


def run_task(
    api_key: str,
    task: str,
    config: ModelConfig,
    *,
    web_mode: str = "auto",
    toa: bool = False,
    toa_workers: int = 2,
    on_content: Callable[[str], None] | None = None,
) -> RunResult:
    """Run one Ya task without choosing a presentation or credential mechanism."""
    client = DeepSeekClient(api_key)
    if toa:
        return toa_agent(client, task, config, toa_workers)
    return single_agent(client, task, config, web_mode=web_mode, on_content=on_content)

File: src/service.ts
import { ModelConfig } from "./config";
import { DeepSeekClient } from "./deepseek";
import { LocalWorkspace } from "./local";
import { singleAgent, toaAgent, type RunResult } from "./orchestrator";
import type { ToolHandler, UserImageContentPart, WebMode } from "./types";

export interface RunTaskOptions {
  webMode?: WebMode;
  toa?: boolean;
  toaWorkers?: number;
  onContent?: (content: string) => void;
  localWorkspace?: LocalWorkspace;
  clientFactory?: (apiKey: string) => DeepSeekClient;
  webSearch?: ToolHandler;
  images?: UserImageContentPart[];
}

export async function runTask(
  apiKey: string,
  task: string,
  config: ModelConfig,
  options: RunTaskOptions = {},
): Promise<RunResult> {
  const client = options.clientFactory?.(apiKey) ?? new DeepSeekClient(apiKey);
  const images = options.images ?? [];
  if (options.toa) return toaAgent(client, task, config, options.toaWorkers ?? 2, options.webSearch, images);
  return singleAgent(
    client,
    task,
    config,
    options.webMode ?? "auto",
    options.onContent,
    options.localWorkspace,
    options.webSearch,
    images,
  );
}
